### ISSUE TITLE
Is it possible to make transparent screen widgets with this method?

### ISSUE THREAD

**Issue by kensully** [Icenowy/Cubway#10]
Hi, congrats on your project. I am impressed with the capability of your project. I tried `Subway Launcher` in my Ubuntu distro, it runs well. I like it  very much. I also tried writing apps with `nwjs`, but it's buggy and it dropped some feature after `chromium` drops `GTK` so it lacks some features in Gnu/Linux. 

I don't know much about `Ċubway` much but it seems to be promising:

Is it possible to make transparent screenlets with it which utilizes `javascript`?

Kind regards.
Kenn

**Comment by 910JQK** [comment]
The capacity of Cubway is based on Qt, which is a excellent library, so it is possible to create screenlet use it if we add a bit of necessary feature to it.
But the problem is, webkit costs a large amount of memory resources. Just making a screenlet or desktop widget is wasteful.
In fact, the initial target of this project is to create a program which covers the whole desktop and runs tile widgets on it. However, due to the consideration of practical value, we finally made a launcher (without widget) which can be taken into foreground at anytime, instead of a desktop keeps in the bottom of all windows.
As for Cubway itself, it is designed to integrate with Linux desktop. But if comparing with NW or Electron, it lacks functionality. With Node, we have a lot of modules and libraries to use directly in JavaScript, but with Qt, there is no available modules, we have to create all kinds of modules (including filesystem, xdg, dbus, etc) to connect JS with C++ and establish the whole ecosystem. It is a difficult task. So when we became busy, the ideal project became obsoleted.

**Comment by kensully** [comment]
Thank you for the info, I understand the mechanism better now. I thought you didn't need to run it on top of a large executable like `nw` which is around 60Mb. I like  `Cubway`s' way of rendering `js` on top of a small executable. I still consider this project valuable because you can distribute the app  within a small package. 
Thanks for all.
Regards,
Kenn

**Comment by 910JQK** [comment]
Yes, the executable is unbelivably small (compare with `nw`). It's an advantage.

Actually I have ever used `nw` to make an IRC client (obsoleted, code maybe nowhere to found). Initially I felt `nw` was an excellent platform, but when I saw its memory cost, I was shocked. There's not only a Webkit but also a Node eating two parts of memory, totally 500MB or so.

On the other hand, the approach of using Qt is tidy and comfort. If you want to get more info about the implementation, is suggested to read [the doc of QtWebKit Bridge](http://qt-project.org/doc/qt-4.8/qtwebkit-bridge.html). Although the `Cubway` project is obsoleted, it is possible to use those techniques to make apps directly. This kind of apps contain two part, JavaScript and C++. If you just want to make **one** app, it is **not** necessary to make all kinds of modules. Only modules related to your app are required. It won't cost much time.

And, sadly, I found I forgot some detailed info in the README doc of `subway launcher`. So I updated the README and added something that may be helpful.
